File: Project_Sourcecode_go/output/strategy/makeDeposit/MakeDeposit1.go
package makeDeposit

import (
	"cs586_project/datastore"
	"fmt"
)

type MakeDeposit1 struct {
}

func (s *MakeDeposit1) MakeDeposit(ds interface{}) {
	ds1 := ds.(*datastore.DS1)
	d := ds1.Temp_d
	b := ds1.Balance
	ds1.Balance = d + b
	fmt.Printf("Make Deposit: %v\n", d)
}

File: Project_Sourcecode_java/src/main/java/edu/iit/cs586/project/output/strategy/MakeDeposit/MakeDeposit1.java
package edu.iit.cs586.project.output.strategy.MakeDeposit;

import edu.iit.cs586.project.datastore.DS1;
import edu.iit.cs586.project.datastore.DataStore;

/**
 * @author CharmingLee <charminglee@jieztech.com>
 * @date 2022/5/2 23:41
 */
public class MakeDeposit1 implements MakeDeposit{
    @Override
    public void makeDeposit(DataStore ds) {
        int d = ((DS1)ds).temp_d;
        int b = ((DS1) ds).b;

        ((DS1) ds).b = d + b;
    }
}
